module3-crud-and-data-models/r1-schema-and-data-models/README.md

```diff
@@ -0,0 +1,124 @@
+## Introduction To Database Schemas
+
+A database schema is a blueprint or architecture of how our data will look. It doesn’t hold data itself, but instead describes the shape of the data and how it might relate to other tables or models.
+
+For example, this is a `customer` schema in MongoDB using Mongoose.
+
+```js
+const mongoose = require("mongoose");
+
+const customer = new mongoose.Schema({
+  name: {
+    type: String,
+    required: true,
+  },
+  zipcode: {
+    type: Number,
+  },
+});
+
+module.exports = mongoose.model("Customer", customer);
+```
+
+The same `customer` schema in a MySql database.
+
+```sql
+CREATE TABLE customer (
+ id INT AUTO_INCREMENT PRIMARY KEY,
+ name VARCHAR(50) NOT NULL,
+ zipcode INT(15) default NULL,
+)
+
+```
+
+We can see that the schema definition for the `customer` has a `name` that is type `string` and a `zipcode` that is of type `number`.
+
+## Introduction to CRUD operations
+
+When we are building APIs, we want our models to provide four basic types of functionality. The model must be able to Create, Read, Update, and Delete resources. Computer scientists often refer to these functions by the acronym CRUD. A model should have the ability to perform at most these four functions in order to be complete.
+
+### CRUD and REST
+
+**What is REST?**
+
+REST is a set of architectural constraints. API developers can implement REST in a variety of ways. When a client request is made via a RESTful API, it transfers data as an object in some format to the requester or endpoint. There are several formats, JSON is the most generally popular file format to use.
+
+**Resource in REST**
+A resource in REST is a like an Object or an Entity in a Database. Once a resource is identified then its representation is to be decided using a standard format so that the server can send the resource in the decided on format and client can understand the same format. A resource can be any information that can be named (e.g. a person, a user, an invoice, a collection of invoices, etc).
```

Comment:
REST is explained in detail in module 1, so this part can be removed.
You could instead rename this section as CRUD in REST, and explain with examples that when handling different REST operations like GET, PUT, POST, DELETE there is most likely an underlying database CRUD operation taking place. Otherwise the examples being shared after this will also be redundant with module 1.

Reply:
After going through module 1, this makes sense, however, I think this process would have been better if more details were provided in the module outline sheet, like _"explain with examples that when handling different REST operations like GET, PUT, POST, DELETE there is most likely an underlying database CRUD operation taking place"_, so I would know easily what to write and explain here.